docs/en/04_Changelogs/4.0.0.md

```diff
@@ -28,12 +85,187 @@ Note that this will not allow you to utilise certain file versioning features in
 
 	:::yaml
 	\SilverStripe\Filesystem\Flysystem\FlysystemAssetStore:
-		legacy_paths: true
+	  legacy_paths: true
 
 
 See [/developer_guides/files/file_management] for more information on how the new system works.
 
 
+### Upgrade code which acts on `Image`
+
+As all image-specific manipulations has been refactored from `Image` into an `ImageManipulations` trait, which 
+is applied to both `File` and `DBFile`. These both implement a common interface `AssetContainer`, which
+has the `getIsImage()` method. In some cases, it may be preferable to invoke this method to detect
+if the asset is an image or not, rather than checking the subclass, as the asset may also be a `DBFile` with
+an image filter applied, rather than an instance of the `Image` dataobject.
+
+In addition, a new file category `image/supported` has been added, which is a subset of the `image` category.
+This is the subset of all image types which may be assigned to the `[api:Image]` dataobject, and may have
+manipulations applied to it. This should be used as the file type restriction on any `[api:UploadField]` which
+is intended to upload images for manipulation. All image categories have also been specified via constants
+on the `[api:File]` DataObject.
```

Comment:
It was actually a suggestion from @sminnee ; I'm also in favour as it adds better code reflection (find usages of particular file categories, etc).

In yml you need to use the literal strings, so this is seen as a helper for this functionality rather than a replacement. The use of constants over strings also helps improve error detection, since typos become parse errors rather than runtime errors.

There's no requirement that a category HAS to be a constant. I disagree that it is harder to add categories. User code could even create their own constants, it just can't be inside the `File` dataobject.